### PR TITLE
feat(libs): add freetype font manager

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Description of the feature or fix

Added freetype Font Manager, features:
1. Reuse font resources requested by the application for fonts with the same parameters.
2. Optimize scenarios where fonts are frequently created to reduce font creation time.
3. font resources can be used in series without manually managing the `font->fallback` pointer.
4. Support custom font path mapping to decouple applications from font resource names.

TODO:
- [ ] Add documents
- [ ] Add an example
- [ ] Add tests

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/others/font_manager/lv_font_manager_utils.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct {
```
Comment (kisvegabor) [inline src/others/font_manager/lv_font_manager_utils.h:30]: Shouldn't it be in an `lv_freetype*` file?

File: src/others/font_manager/lv_font_manager.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+typedef struct _lv_font_manager_t lv_font_manager_t;
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/**
+ * Create main font manager.
+ * @param recycle_cache_size size of the recycle cache.
```
Comment (kisvegabor) [inline src/others/font_manager/lv_font_manager.h]: Explain what "recycle cache" is and the tell its unit (byte, pieces of fonts?)

File: src/others/font_manager/lv_font_manager.h [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ */
+void lv_font_manager_add_path(lv_font_manager_t * manager, const char * name, const char * path);
+
+/**
+ * Add the font file path with static memory.
+ * @param manager pointer to main font manager.
+ * @param name font name.
+ * @param path font file path.
+ */
+void lv_font_manager_add_path_static(lv_font_manager_t * manager, const char * name, const char * path);
```
Comment (kisvegabor) [inline src/others/font_manager/lv_font_manager.h:66]: Name is usually short and I assume that there will be only a few font_manager. Is it really required to have a `*_static()` variant?

**Review by kisvegabor — COMMENTED** [review]
Is there a way to support font engines in a abstract way? E.g. font engines can provide the required functions or register callback. The goal would be to make both FreeType and TinyTTF supported in a modular way.

**Comment by kisvegabor** [comment]
It would be nice to make it really abstract by hiding static fonts (e.g. lv_font_monsterrat_10/12/14...48) behind the font manager too.


cc @richardgazdik @coevin : It can be useful for the editor's font manager too.

File: src/others/font_manager/lv_font_manager.h (hunk 3 above)
Comment (FASTSHIFT) [reply src/others/font_manager/lv_font_manager.h:66]: The font file path is usually long, often reaching more than 100 characters.

Comment (FASTSHIFT) [reply src/others/font_manager/lv_font_manager.h:66]: There are often more than 10 font files in a product, and using `static` makes sense.

File: src/others/font_manager/lv_font_manager_utils.h (hunk 1 above)
Comment (FASTSHIFT) [reply src/others/font_manager/lv_font_manager_utils.h:30]: This data structure is only for font_manager's internal use at the moment and can be incorporated into lv_freetype at a later stage.

File: src/others/font_manager/lv_font_manager.h (hunk 2 above)
Comment (FASTSHIFT) [reply src/others/font_manager/lv_font_manager.h]: The unit of cache is the number of fonts (lv_font_t).

**Comment by kisvegabor** [comment]
What's the plan with this feature?

**Comment by FASTSHIFT** [comment]
> What's the plan with this feature?

While we are busy with project delivery. how about we first implement a freetype manager that only supports ?

**Comment by kisvegabor** [comment]
It's ok for me, but promise that it will be updated before v9.3 :smile: 
I've added an entry for it [here](https://github.com/lvgl/lvgl/issues/6763)

**Review by kisvegabor — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Comment by FASTSHIFT** [comment]
Continue here: https://github.com/lvgl/lvgl/pull/7058

**Pull request merged**